libs/store_helper/ebpf_store_helper.c

```diff
@@ -144,6 +149,11 @@ ebpf_store_update_section_information(
     for (uint32_t i = 0; i < section_info_count; i++) {
         ebpf_store_key_t section_key = NULL;
 
+        if (!ebpf_validate_program_section_info(&section_info[i])) {
```

Comment:
I don't think it means that.  Multiple versions can be accepted as valid at the same time.

Reply:
No @dthaler - not quite. eBPF v1.N can update an extension struct and maintain backward compatibility with the structs shipped in versions v1.0 - v1.N-1 as by definition such changes will be backward compatible. However, once there is a change in the struct that is no longer backward compatible (such as adding a field in the middle or changing the type of a field), then we must change the version of the struct and will be considered a breaking change. We will then have to bump the version of the product to v2.0.

Reply:
Once you bump the version of a structure, it doesn't mean that code cannot accept two versions of the same structure.  It can.  Thus a newer version of ebpfapi.dll can support an older extension.   My hope though is that we never need to bump the version.